sql-plugin/src/main/scala/com/nvidia/spark/rapids/profiler.scala

```diff
@@ -119,9 +122,27 @@ object ProfilerOnExecutor extends Logging {
       val stageId = taskCtx.stageId
       if (stageRanges.contains(stageId)) {
         synchronized {
-          activeStages.add(taskCtx.stageId)
-          enable()
-          startPollingDriver()
+          if (taskLimit <= 0) {
+            // Unlimited tasks per stage
+            activeStages.add(taskCtx.stageId)
+            enable()
+            startPollingDriver()
+          } else {
+            // Limited tasks per stage
+            if (stageTaskCount.getOrElse(stageId, 0) < taskLimit) {
+              activeStages.add(taskCtx.stageId)
+              enable()
+              startPollingDriver()
+            }
+            taskCtx.addTaskCompletionListener[Unit] { _ =>
+              val currentCount = stageTaskCount.getOrElse(stageId, 0)
+              if (currentCount < taskLimit) {
+                stageTaskCount(stageId) = currentCount + 1
+              } else {
+                disable()
```

Comment:
Just because a stage has completed a task does not mean we want to disable profiling.  Tasks for other stages/jobs that have been flagged for profiling could be active, and we want to keep profiling enabled in that case.  See the logic in updateActiveFromDriver.  Instead, this needs to do something like the following. Instead of copying this code, it should be refactored from the version in updateActiveFromDriver and reused.
```scala
                synchronized {
                  if (activeJobs.isEmpty && activeStages.isEmpty) {
                    disable()
                    stopPollingDriver()
                  }
                }
```

Reply:
Updated `updateActiveFromDriver` to check if we want to disable profiling. The current code only disables profiling, but does not stop polling driver in the case of task limiting, because the `allDone` status updating in `onStageCompleted` from the driver side, so it would be a little tricky to stop polling driver earlier when a stage is not yet complete.